BraveShareTo/ShareToBraveViewController.swift

```diff
@@ -0,0 +1,110 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Social
+import MobileCoreServices
+
+class ShareToBraveViewController: SLComposeServiceViewController {
+    
+    // TODO: Separate scheme for debug builds, so it can be tested without need to uninstall production app.
+    private func urlScheme(for url: String) -> NSURL? {
+        return NSURL(string: "brave://open-url?url=\(url)")
+    }
+    
+    override func configurationItems() -> [Any]! {
+        guard let inputItems = extensionContext?.inputItems as? [NSExtensionItem] else {
+            return []
+        }
+        
+        // Reduce all input items down to a single list of item providers
+        let attachments: [NSItemProvider] = inputItems
+            .compactMap { $0.attachments as? [NSItemProvider] }
+            .flatMap { $0 }
```

Comment:
This is deprecated and redundant. I think this was just copied, but should remove `flatMap`

Reply:
This is actually stuff I added, and `flatMap` isn't deprecated in this usage :) It was replaced with `compactMap` for the usage of mapping `[T?]` to `[T]`.

This usage of `flatMap` takes `[[T]]` and flattens it to `[T]`

Reply:
Oh interesting, didn't see it in doc reference, nice 😃